BREAKING.md

```diff
@@ -12,10 +12,18 @@ There are a few steps you can take to write a good change note and avoid needing
 
 ## 0.54 Breaking changes
 - [Removed `readAndParseFromBlobs` from `driver-utils`](#Removed-readAndParseFromBlobs-from-driver-utils)
+- [Removed `innerRequestHandler`](#Removed-innerRequestHandler)
 
 ### Removed `readAndParseFromBlobs` from `driver-utils`
 The `readAndParseFromBlobs` function from `driver-utils` was deprecated in 0.44, and has now been removed from the `driver-utils` package.
 
+### Removed `innerRequestHandler`
+`innerRequestHandler` is removed from `@fluidframework/request-handlers` package, and its usage is removed from `BaseContainerRuntimeFactory` and `ContainerRuntimeFactoryWithDefaultDataStore`.  If you are using these container runtime factories, attempting to access internal data stores via `request()` will result in 404 responses.
+
+If you rely on `request()` access to internal root data stores, you can add `rootDataStoreRequestHandler` to your list of request handlers on the runtime factory.
```

Comment:
I don't think we should call out using `rootDataStoreRequestHandler`. Ideally, consumers should directly call `getRootDataStore` API and not rely on these handlers.
Even if we do decide to recommend using `rootDataStoreRequestHandler`, we should do so when we have the right patterns defined.

Reply:
Maybe let's append a sentence to this paragraph to the effect of "It is not recommended to allow generic access to root data stores via `request()` -- instead prefer to retrieve a well-known root data store and selectively provide access to further data stores through its API surface"?

`getRootDataStore()` is great from inside the container, but for hosts trying to retrieve container content from the outside we need to have at least some recommendation on how to proceed.

Reply:
To wrap up this PR we could go step by step:

Remove `innerRequestHandler`, while offering guidelines on the best practices @vladsud and @agarwal-navin mentioned. We can still call out that `rootDataStoreRequestHandler` is available as a last resort (unbreak yourself), but at the same time we can mark it deprecated now to discourage usage.

In future release we can remove `rootDataStoreRequestHandler` It is not widely used anyways. 

General question @agarwal-navin: If this is not the best practice, is this just old code in Bohemia that still relies on request handler in BaseComponent ?